tests/composition/test_composition.py

```diff
@@ -4342,18 +4343,18 @@ def test_scheduler_conditions(self, mode, condition, expected_result):
         response = pnl.ProcessingMechanism(size=2, name="GATE")
 
         comp = pnl.Composition()
-        comp.add_node(decisionMaker)
-        comp.add_node(response)
-        comp.add_projection(pnl.MappingProjection(), sender=decisionMaker, receiver=response)
-
-        if condition is pnl.EveryNCalls:
-            comp.scheduler.add_condition(response, condition(decisionMaker, 5))
-        elif condition is pnl.BeforeNCalls:
-            comp.scheduler.add_condition(response, condition(decisionMaker, 5))
+        comp.add_linear_processing_pathway([decisionMaker, response])
+
+        if condition is pnl.BeforeNCalls:
+            comp.scheduler.add_condition(response, condition(decisionMaker, 5,
+                                                             time_scale=scale))
         elif condition is pnl.AtNCalls:
-            comp.scheduler.add_condition(response, condition(decisionMaker, 5))
+            comp.scheduler.add_condition(response, condition(decisionMaker, 5,
+                                                             time_scale=scale))
         elif condition is pnl.AfterNCalls:
-            comp.scheduler.add_condition(response, condition(decisionMaker, 5))
+            c = 1 if scale is TimeScale.PASS else 5
```

Comment:
this could use a comment explaining why c depends on timescale

Reply:
Added a comment. Mechanisms run only once per PASS unless they're in `run_until_finished`.